### PR TITLE
New assembly and modules for CNV-1616 Editing VMs

### PR DESCRIPTION
**Pull request by bgaydosrh**
@openshift/team-documentation

This change is targeted for 2.0.

The module cnv-editing-vm-cli.adoc likely needs a SME review. If you can recommend a reviewer, please let me know.

Thanks
Bob

### CODE REVIEW

**Comment by vikram-redhat** [comment]
@bgaydosrh we will wait for the QE reviewer to provide the review for this. Marking this for just peer review.

File: cnv/cnv_users_guide/cnv-edit-vms.adoc [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[id="cnv-edit-vms"]
+= Editing virtual machines
+include::modules/cnv_document_attributes.adoc[]
+:context: cnv-edit-vms
+toc::[]
+
+Edit a virtual machine by doing one of the following:
```
Comment (kalexand-rh) [inline cnv/cnv_users_guide/cnv-edit-vms.adoc]: ~doing~ completing one of the following tasks:

Comment (bgaydosrh) [reply cnv/cnv_users_guide/cnv-edit-vms.adoc]: Done.

**Comment by bgaydosrh** [comment]
Squashed and re-pushed to 1 commit.

File: modules/cnv-editing-vm-yaml-web.adoc [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+====
+Navigating away from the YAML screen while editing cancels any changes to the configuration you have made.
+====
+
+.Procedure
+
+. Click *Workloads* -> *Virtual Machine* from the side menu.
+. Select a virtual machine.
+. Click the *YAML* tab to display the editable configuration.
+.. (Optional) You can click *Download* to download the YAML file locally in its current state.
```
Comment (kalexand-rh) [inline modules/cnv-editing-vm-yaml-web.adoc]: why is this a substep?
Optional steps use the “. Optional: <step_text>.” format.

File: modules/cnv-editing-vm-yaml-web.adoc [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+. Edit the file and click *Save*.
+
+A confirmation message shows that the modification has been successful, including the updated version number for the object.
```
Comment (kalexand-rh) [inline modules/cnv-editing-vm-yaml-web.adoc]: s/successful, including/successful and includes

File: modules/cnv-editing-vm-cli.adoc [hunk 4]
```diff
@@ -0,0 +1,21 @@
+// Module included in the following assemblies:
+//
+// * cnv/cnv_users_guide/cnv-edit-vms.adoc
+
+[id="cnv-editing-vm-cli_{context}"]
+
+[[cnv-editing-vm-cli]]
```
[the remark this answers is not in the document]
Comment (bgaydosrh) [reply modules/cnv-editing-vm-cli.adoc]: Changed to: 
[id="cnv-editing-vm-cli_{context}"]
= Using the CLI to edit a virtual machine

File: modules/cnv-editing-vm-cli.adoc [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+= Using the CLI to edit a virtual machine
+
+.Prerequisites
+
+* Your virtual machine must have been initially configured with a YAML object configuration file.
```
[the remark this answers is not in the document]
Comment (bgaydosrh) [reply modules/cnv-editing-vm-cli.adoc]: Changed to: 
.Prerequisites

* You configured your virtual machine with a YAML object configuration file.
* You installed the oc CLI.

I will add the question on clarity to the questions for QE.

File: modules/cnv-editing-vm-cli.adoc [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+.Procedure
+
+. Run `oc edit` to update the virtual machine configuration.
```
[the remark this answers is not in the document]
Comment (bgaydosrh) [reply modules/cnv-editing-vm-cli.adoc]: Done.

File: modules/cnv-editing-vm-cli.adoc [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+. Open the object configuration.
+. Edit the YAML.
+
+NOTE: If you edit a running virtual machine, you need to either restart the virtual machine or run `oc apply` for the new configuration to take effect.
```
[the remark this answers is not in the document]
Comment (bgaydosrh) [reply modules/cnv-editing-vm-cli.adoc]: Done.

File: modules/cnv-editing-vm-web.adoc [hunk 8]
```diff
@@ -0,0 +1,26 @@
+// Module included in the following assemblies:
+//
+// * cnv/cnv_users_guide/cnv-edit-vms.adoc
+
+[id="cnv-editing-vm-web_{context}"]
+
+= Using the web console to edit a virtual machine
+
+Edit select values of a virtual machine in the web console or from the *Virtual Machine Overview* screen. Other values can be edited using the CLI.
```
[the remark this answers is not in the document]
Comment (bgaydosrh) [reply modules/cnv-editing-vm-web.adoc]: Recording this for QE and other team members to address at a later date. We're out of time now.

File: modules/cnv-editing-vm-web.adoc [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+.Procedure
+
+. Click *Workloads* -> *Virtual Machines* from the side menu.
+. Select a Virtual Machine.
+. Click *Edit* to make editable fields available.
+. You can change the *Flavor* but only to *Custom*, which provides additional fields for *CPU* and *Memory*.
+. Click *Save*.
+
+The updated values are shown after the operation is processed.
```
[the remark this answers is not in the document]
Comment (bgaydosrh) [reply modules/cnv-editing-vm-web.adoc]: Recording for QE to address.

File: modules/cnv-editing-vm-yaml-web.adoc [hunk 10]
```diff
@@ -0,0 +1,30 @@
+// Module included in the following assemblies:
+//
+// * cnv/cnv_users_guide/cnv-edit-vms.adoc
+
+[id="cnv-editing-vm-yaml-web_{context}"]
+
+= Editing the virtual machine YAML configuration
+
+Edit the YAML configuration of a virtual machine directly within the web console.
```
[the remark this answers is not in the document]
Comment (bgaydosrh) [reply modules/cnv-editing-vm-yaml-web.adoc]: Removed.

File: modules/cnv-editing-vm-yaml-web.adoc [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+
+Not all parameters can be updated. If you edit values that cannot be changed and click *Save*, an error message indicates the parameter that was not able to be updated.
```
[the remark this answers is not in the document]
Comment (bgaydosrh) [reply modules/cnv-editing-vm-yaml-web.adoc]: Recording this for QE and other team members to address at a later date.

File: modules/cnv-editing-vm-yaml-web.adoc (hunk 2 above)
Comment (bgaydosrh) [reply modules/cnv-editing-vm-yaml-web.adoc]: Done.

File: modules/cnv-editing-vm-yaml-web.adoc (hunk 3 above)
Comment (bgaydosrh) [reply modules/cnv-editing-vm-yaml-web.adoc]: Done.

**Pull request merged**